pkg/storage/swift/swift.go

```diff
@@ -348,33 +353,49 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
 		return fmt.Errorf("failed to get cluster infrastructure info: %v", err)
 	}
 
-	// Generate new container name if it wasn't provided.
-	// The name has cluster ID as a prefix plus "image-registry" suffix, which is complemented
-	// by 8 capital latin letters
-	// Example of a generated name: user-298xw-image-registry-FHEIBGDD
-	if cr.Spec.Storage.Swift.Container == "" {
-		cr.Spec.Storage.Swift.Container = generateContainerName(infra.Status.InfrastructureName)
-	}
+	generatedName := false
+	for i := 0; i < 5000; i++ {
+		if len(cr.Spec.Storage.Swift.Container) == 0 {
+			if cr.Spec.Storage.Swift.Container, err = util.GenerateStorageName(d.Listers, ""); err != nil {
+				return err
+			}
+			generatedName = true
+		}
 
-	createOps := containers.CreateOpts{
-		Metadata: map[string]string{
-			"Openshiftclusterid": infra.Status.InfrastructureName,
-			"Name":               cr.Spec.Storage.Swift.Container,
-		},
-	}
+		if err = d.containerExists(client, cr.Spec.Storage.Swift.Container); err != nil {
+			if _, ok := err.(*gophercloud.ErrResourceNotFound); ok {
+				if !generatedName {
+					util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapi.ConditionFalse, "Unable to access container", "The user provided container does not exist")
+					break
+				}
+			} else {
+				cr.Spec.Storage.Swift.Container = ""
```

Comment:
Let's see if I am getting this right. If the user provides a container name and we fail to check if the container exists(by any other reason than `gophercloud.ErrResourceNotFound`) we gonna ended up creating a new one with a new name. Is that right?

Also just for my understanding, if the container does exist why do we attempt to recreate it? I would assume that if the container exists we would just say: "OK, it is created".

Reply:
Maybe it is by design but I added the following "test":

```
func TestInternalServerError(t *testing.T) {                                                                                                        
        th.SetupHTTP()                                                                                                                  
        defer th.TeardownHTTP()                                                                                                         
        handleAuthentication(t, "container")                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                 
        th.Mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
                fmt.Println(r.URL)                                                                                                                                                                                                    
                w.WriteHeader(500)                                                                                                                                                                                                                                               
        })                                                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                                                 
        d, installConfig := mockConfig(false, th.Endpoint()+"v3", MockUPISecretNamespaceLister{})                                                                                                                                                                                
        d.CreateStorage(&installConfig)                                                                                                 
}
```

What I noticed is that the endpoint is called 5000 times, on the first call the url is `/registry`, all other 4999 are something like `/user-j45xj-image-registry--gmwjgsapgatlmodylumguqdiwwmnppqimge`, that makes me believe that if the remote server is crashed we may have other problems.

Reply:
It is only called the other 4,999 times if the names that we generate are taken.

Reply:
@coreydaley So, if user provided `X` as the container name and we fail to check if `X`  exists(like the `500` error I used on my test) shall we go ahead and generate a new random name?

I am trying to understand if that is the expected behavior, strange things can happen when we have a network connection involved :-)

Reply:
Ah, sorry. I commented before noting you have already changed the implementation. LGTM now.